app/vmui/packages/vmui/src/utils/time.ts

```diff
@@ -26,6 +26,22 @@ const shortDurations = supportedDurations.map(d => d.short);
 
 export const roundToMilliseconds = (num: number): number => Math.round(num*1000)/1000;
 
+const roundStep = (step: number) => {
+  if (step >= 100) {
+    return Math.round(step) - (Math.round(step)%10); // step 10
+  }
+  if (step < 100 && step >= 10) {
+    return Math.round(step) - (Math.round(step)%5); // step 5
+  }
+  if (step < 10 && step >= 1) {
+    return Math.round(step); // step 1
+  }
+  if (step < 1 && step > 0.01) {
+    return Math.round(step * 40) / 40; // step 0.025
```

Comment:
there is not step 0.25 it would be the same value.

Reply:
I'm not sure how to properly describe e.g. thousandths multiple of 5

example rounding: 
```
"0.04168297809524568 to 0.05"
"0.9717025629494032 to 0.975"
"0.9432368293005493 to 0.95"
"0.35041777044984834 to 0.35"
"0.854946547243459 to 0.85"
"0.08957464505352353 to 0.1"
"0.9590827495368304 to 0.95"
"0.1843230483492937 to 0.175"
"0.8327118230614492 to 0.825"
"0.5955052834181376 to 0.6"
```

Reply:
I'm unsure this fancy rounding is needed at all - it was enough rounding the step to milliseconds like it was before, since:

* VictoriaMetrics accepts `step` value with millisecond precision
* This is easier to debug than the fancy rounding
* Grafana also rounds the step to milliseconds as I know

Which issue the fancy rounding of the step does resolve? @hagen1778 , wdyt?

Reply:
Oh, I discovered the one legitimate case for fancy rounding - it makes response caching at VictoriaMetrics side more efficient, since the `step` value is used as a part of the response cache. So if the `step` value changes, then all the cached response results for the previous `step` values aren't used. This rounding will increase chances that the cached responses will be used when the user changes zoom level.